### ISSUE TITLE
`common.features.infer_types.check_if_nlp_feature` fails on bytes columns

### ISSUE THREAD

**Issue by samg-stripe** [autogluon/autogluon#4602]
https://github.com/autogluon/autogluon/blob/ca3e0b5cadb064e256cd836b4214046aefae66bd/common/src/autogluon/common/features/infer_types.py#L141-L144

This try/catch only handles `AttributeError`, but if a sequence of bytes is passed, a `TypeError` occurs due to `.str.split()`

**Comment by rsj123** [comment]
Here is a sample code
``` python
import pandas as pd
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor
from autogluon.tabular import FeatureMetadata


data = {'text_column': [b'hello world', b'this is a test', b'another example']}
df = pd.DataFrame(data)

feature_metadata = FeatureMetadata.from_df(df)

print(feature_metadata)
```

**Comment by Innixma** [comment]
Thanks for highlighting this issue @samg-stripe and thanks @rsj123 for providing an example and a fix PR in #4682! #4682 has been merged, so I will resolve this issue, cheers!
